src/utils/utils-vega.js

```diff
@@ -190,13 +196,23 @@ export function createVegaAttrMixin(
           }
           domainVals = layerObj[domainGetterFunc]()
         }
-        if (capAttrObj.type === "ordinal") {
+        if (scaleType === "ordinal") {
           ordScale.domain(domainVals).range(capAttrObj.range)
           // if color range is not in domain, it's an Other item
           rtnVal =
             domainVals.indexOf(input) === -1
               ? ordScale("Other")
               : ordScale(input)
+        } else if (scaleType === "nominal") {
+          let formattedInput = input
```

Comment:
can you describe the use case here as it pertains to needing to convert an input to a boolean? I understand that if all the domains for the scale are boolean, then the input value needs to be boolean, but it feels like you're protecting against a special use case here. I guess what I'm asking is, what's the use case you've observed where the `input` is not a boolean, and therefore needs to be converted, when the `domain` values are boolean (implying at most a true and false domain)?

Reply:
The special case _was_ when the vega lite like spec was passing true/false for the domain, but the actual values returned from the query were 0/1... not sure if this is the case anymore, digging in a bit